python/graphstorm/config/argument.py

```diff
@@ -2481,6 +2494,12 @@ def _add_lm_model_args(parser):
             help="Whether to cache the LM embeddings in files. " + \
                     "If the LM embeddings have been saved before, load the saved embeddings " + \
                     "instead of computing the LM embeddings again.")
+    group.add_argument("--use-wholegraph-cache-lm-embed",
```

Comment:
do we need an argument to turn on this? if a user uses wholegraph, why is it not turned on automatically?

Reply:
>if a user uses wholegraph, why is it not turned on automatically?

To clarify, do you mean as long as WholeGraph is installed/available on user-end, we turn on this feature automatically? Note that WholeGraph has limitations and might not work in certain scenarios (for now), e.g., (1) needs all procs alive/participate in its operations (dist_tensor creation/gather/scatter) if using the default NCCL backend; (2) needs all files accessible by all machines/procs when loading into WholeGraph memory. However, distDGL (default) is more flexible.

On the other hand, there are many wholegraph-related options/functionalities currently in gs (`feature store`, `sparse embedding`, this PR, etc). At some point, I think we should have a unified flag to turn on them all to improve usability.

Reply:
there is a flag that can turn on wholegraph support in graphstorm. i don't think we need a specific flag to turn on wholegraph for LM cache.

Reply:
if wholegraph support is turned on, we should enable wholegraph for every possible component in graphstorm.

Reply:
> there is a flag that can turn on wholegraph support in graphstorm. 

Which flag/argument? I traced back to the first PR of adding WholeGraph for feature store https://github.com/awslabs/graphstorm/pull/405. It seems it turned on WholeGraph implicitly by file name pattern matching, i.e.,:
https://github.com/awslabs/graphstorm/blob/main/python/graphstorm/utils.py#L102-L108 

And this PR: https://github.com/awslabs/graphstorm/pull/677  uses an argument `--use-wholegraph-sparse-emb` to turn on WholeGraph for sparse embedding.

Reply:
>if wholegraph support is turned on, we should enable wholegraph for every possible component in graphstorm.

I agree. So, maybe we can (1) keep the way of using WholeGraph for inital feature store  as https://github.com/awslabs/graphstorm/pull/405 ; and (2) combine all other WholeGraph componetns for intermediate distributed store in one flag/argument.  Does it sound reasonable? @zheng-da @classicsong 

If so, I will change the flag name in this PR and have another PR to consolidate the wholegraph-related flags.

Reply:
Done the part in this PR. I will open another PR to let sparse_emb use the same argument.